lib/db.js

```diff
@@ -84,7 +93,62 @@ class LimitDBRedis extends EventEmitter {
     });
   }
 
+  #setupPing() {
+    if (!this.pingConfig?.enabled) {
+      return;
+    }
+
+    this.failedPings = 0;
+    this.once('ready', () => {
+      this.#startPing();
+    })
+  }
+
+  #startPing() {
+    this.keepPingRunning = true;
+    const doPing = () => {
+      let hrstart = process.hrtime();
+      this.redis.ping((err, payload) => {
+        let hrend = process.hrtime(hrstart);
+        let duration = hrend[0]*1000 + hrend[1]/1000000
+        err ? this.#pingKO(err, duration) : this.#pingOK(duration);
+
+        if(this.keepPingRunning) {
+          this.pingIntervalId = setTimeout(doPing, this.pingConfig.interval);
+        }
```

Comment:
What do you think, if the ping has being breach i wont re-set the timeout and let `ready` to re-start the setup?

Reply:
Mmm. I thought about that too, but the problem I saw is that we'll need to keep this running retrying reconnects until it recovers (that's why I introduced the `maxFailedAttemptsToRetryReconnect`). 
Picture a network glitch where we lose the connection for a minute; after 10 seconds we issue a reconnect, but it will get lost; it will only reconnect on the first reconnect attempt after the connection gets re-established (~10 reconnect attempts).

Reply:
So, what I'm trying to say is that we need to issue as many reconnects (`redis.disconnect(true)`) as needed, as we will never know which will be the one doing the actual reconnection until we receive the `ready` event.